src/platform/elements/date-picker/DatePickerInput.ts

```diff
@@ -95,46 +91,75 @@ export class NovoDatePickerInputElement implements OnInit, ControlValueAccessor
 
   _handleInput(event: KeyboardEvent): void {
     if (document.activeElement === event.target) {
-      let value = (event.target as HTMLInputElement).value;
-      try {
-        let dateTimeValue = Date.parse(value);
-        if (!isNaN(dateTimeValue)) {
-          let dt = new Date(dateTimeValue);
-          this.dispatchOnChange(dt);
-        } else {
-          this.dispatchOnChange(null);
-        }
-      } catch (err) {}
-      this.openPanel();
+      this._handleEvent(event, false);
     }
   }
 
+  _handleBlur(event: FocusEvent): void {
+    this._handleEvent(event, true);
+  }
+
+  _handleEvent(event: Event, blur: boolean): void {
+    let value = (event.target as HTMLInputElement).value;
+    try {
+      let dateTimeValue = Date.parse(value);
+      if (!isNaN(dateTimeValue)) {
+        let dt = new Date(dateTimeValue);
+        if (dt !== this.value) {
```

Comment:
This logic is duplicated is dispatchOnChange too.  I think we can clean the flow up more